### PR TITLE
Update firebase auth readme.

### PR DESCRIPTION
**Pull request by branflake2267**
What do you think handling this by a future?

### CODE REVIEW

File: packages/firebase_auth/README.md [hunk 1]
```diff
@@ -59,21 +59,26 @@ final FirebaseAuth _auth = FirebaseAuth.instance;
...
+  print("signed in " + user.displayName);
+  return user;
 }
 ```
 
+Then from the sign in button on press, 
+call the `_handleSignIn` method using a future callback for both the `FirebaseUser` and possible exception.
+```
+_handleSignIn()
+    .catchError((e) => print(e));
```
Comment (sethladd) [inline packages/firebase_auth/README.md]: remove semicolon

File: packages/firebase_auth/README.md [hunk 2]
```diff
@@ -59,21 +59,26 @@ final FirebaseAuth _auth = FirebaseAuth.instance;
...
+    .then((FirebaseUser user) => print(user))
```
Comment (sethladd) [inline packages/firebase_auth/README.md]: semicolon here?

Comment (sethladd) [inline packages/firebase_auth/README.md]: actually, it's then() and then catchError()

Comment (branflake2267) [inline packages/firebase_auth/README.md]: oops, I was free handing, good catch.

**Comment by sethladd** [comment]
cc @kroikie

**Comment by kroikie** [comment]
Seems reasonable, also a bit more darty.

File: packages/firebase_auth/README.md [hunk 3]
```diff
@@ -59,21 +59,26 @@ final FirebaseAuth _auth = FirebaseAuth.instance;
...
-    print(error);
-  }
+Future<FirebaseUser> _handleSignIn() async {
+  GoogleSignInAccount googleUser = await _googleSignIn.signIn();
+  GoogleSignInAuthentication googleAuth = await googleUser.authentication;
+  FirebaseUser user = await _auth.signInWithGoogle(
+    accessToken: googleAuth.accessToken,
+    idToken: googleAuth.idToken,
+  );
+  print("signed in " + user.displayName);
```
Comment (sethladd) [inline packages/firebase_auth/README.md]: Next time, you can use string interpolation: `"signed in ${user.displayName}"`

**Pull request merged**